File: Others/BinaryToInteger/BinaryToIntegerv1.java
package BinaryToInteger;

public class BinaryToIntegerv1 {
    public static void main(String[] args) {
        String binaryString;
        int decimalValue;

        try (java.util.Scanner kbd = new java.util.Scanner(System.in)) {
            do {
                System.out.print("Enter a binary number (or 'exit' to quit): ");
                binaryString = kbd.nextLine();

                if (!binaryString.equalsIgnoreCase("exit")) {
                    try {
                        decimalValue = Integer.parseInt(binaryString, 2);
                        System.out.println("The decimal value of " + binaryString + " is: " + decimalValue);
                    } catch (NumberFormatException e) {
                        System.out.println("Invalid binary number: \"" + binaryString + "\" is not a valid binary number.");
                    }
                }
            } while (!binaryString.equalsIgnoreCase("exit"));
        }
        System.out.println("Program terminated. Goodbye!");
    }
}

File: Others/BinaryToInteger/BinaryToIntegerv1.py
def BinaryToInteger(n):
    """
    Convert a binary number (as a string) to an integer.
    
    Args:
    n (str): A string representing a binary number.
    
    Returns:
    int: The integer value of the binary number.
    
    Raises:
    ValueError: If the input string is not a valid binary number.
    """
    # Check if the input is a valid binary number
    if not all(char in '01' for char in n):
        raise ValueError("Input must be a binary number (string of 0s and 1s).")
    
    # Convert binary string to integer
    return int(n, 2)

while True:
    n = int(input("Enter a binary number (or -1 to exit): "))
    if n == -1:
        break
    try:
        result = BinaryToInteger(str(n))
        print(f"The integer value of binary {n} is: {result}")
    except ValueError as e:
        print(n + " is not a valid binary number.")
